### PR TITLE
re-org docker file, upgrades

### PR DESCRIPTION
**Pull request by jharenza**
#### What scientific question is your analysis addressing?
This PR upgrades a few python packages due to newly introduced bugs in the consensus CN modules

#### What was your approach?
python 3.10.12-->3.11
snakemake 7.25.3-->8.1.2
pulp ? --> 2.8.0
install MM2S from GitHub

#### Reproducibility Checklist

- [ ] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [ ] This analysis has been added to continuous integration.

#### Documentation Checklist

- [ ] This analysis module has a `README` and it is up to date.
- [ ] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [ ] The analytical code is documented and contains comments.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -6,47 +6,134 @@ ARG github_pat=$GITHUB_PAT
 
 ENV GITHUB_PAT=$github_pat
 
+# Avoid warnings by switching to noninteractive
+ENV DEBIAN_FRONTEND=noninteractive
```
Comment (devbyaccident) [inline Dockerfile:10]: This is a good change. :)

File: Dockerfile [hunk 2]
```diff
@@ -193,83 +279,27 @@ RUN R -e "remotes::install_github('jokergoo/circlize', ref = 'b7d86409d7f893e881
...
-    "utils==1.0.1" \
-    "webencodings==0.5.1" \
-    "widgetsnbextension==4.0.7" \
-    "wrapt==1.15.0" \
-    "zipp==3.15.0" \
-    "openpyxl==3.1.2" \
-    && rm -rf /root/.cache/pip/wheels
+# Packages required for de novo mutational signatures
+#RUN install2.r --error --deps TRUE \
+#    lsa
```
Comment (devbyaccident) [inline Dockerfile:290]: Were these two lines meant to be commented out?

**Review by devbyaccident — APPROVED** [review]
LGTM, just threw in a couple clarifying questions/comments.

File: Dockerfile (hunk 2 above)
Comment (jharenza) [reply Dockerfile:290]: Yes, not using this package at the moment

**Pull request merged**